src/common/storeKey.ts

```diff
@@ -0,0 +1,28 @@
+export interface IStoreKey<T> {
```

Comment:
If two unrelated modules decide to provide `IStoreKey<T>` instances with the same name, but expect different types, there could be collisions. A way to mitigate this would be to use an internal auto-incremented `id` property for the mapping. `name` would just be for the error message.